lms/djangoapps/instructor/views/legacy.py

```diff
@@ -730,14 +745,22 @@ def get_student_grade_summary_data(
                     # value as a float between 0 and 1, or a two-tuple
                     # containing the earned score and possible score for
                     # the assignment (see docstring).
-                    for score in gradeset['raw_scores']:
-                        if get_score_max is True:
-                            add_grade(score.section, score.earned, score.possible)
-                        else:
-                            add_grade(score.section, score.earned)
+                    for chapter in progress_summary.chapters:
+                        for section in chapter['sections']:
+                            earned = float(section['section_total'].earned)
+                            possible = float(section['section_total'].possible)
+                            label = section['section_total'].section
+                            add_grade(label, earned, possible=possible)
                 else:
-                    for grade_item in gradeset['section_breakdown']:
-                        add_grade(grade_item['label'], grade_item['percent'])
+                    for chapter in progress_summary.chapters:
+                        for section in chapter['sections']:
+                            earned = section['section_total'].earned
+                            possible = section['section_total'].possible or 1
+                            label = section['section_total'].section
+                            percent = earned / float(possible)
+                            percent = round(percent * 100 + 0.05) / 100
+                            add_grade(label, percent, possible=1)
+
```

Comment:
I think this can be more DRY, since the only difference between the branches is the calculation of the percentages and the call to add_grade()